dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/engine/NotificationRuleActionImplementer.java

```diff
@@ -56,10 +58,16 @@
     // -------------------------------------------------------------------------
 
     @Autowired
-    private ProgramNotificationTemplateStore programNotificationTemplateStore;
+    protected ProgramNotificationTemplateStore programNotificationTemplateStore;
 
     @Autowired
-    private NotificationLoggingService notificationLoggingService;
+    protected NotificationLoggingService notificationLoggingService;
```

Comment:
Can you use   constructor based injection, so we keep the same approach everywhere